mitmproxy/protocol/http2.py

```diff
@@ -96,15 +96,16 @@ def __init__(self, ctx, mode):
         self.server_to_client_stream_ids = dict([(0, 0)])
         self.client_conn.h2 = SafeH2Connection(self.client_conn, client_side=False, header_encoding=False)
 
-        # make sure that we only pass actual SSL.Connection objects in here,
-        # because otherwise ssl_read_select fails!
-        self.active_conns = [self.client_conn.connection]
-
```

Comment:
`__call__` started with `if self.server_conn`, which is now missing. Truth to be told, we never hit different code paths as long as `protocol.tls` always connects upstream because of ALPN (ugh). Whatsoever, that is/was the reason why we have that distinction into `active_conns`. So, the whole story is pretty bad and we probably need to do something about it at some point. I'm a bit in a disagreement with the changes here, so maybe let's leave that out so that we don't get sidetracked?